envi/tests/test_arch_arm.py

```diff
@@ -5,68 +5,577 @@
 import envi.registers as e_reg
 import envi.memcanvas as e_memcanvas
 import envi.memcanvas.renderers as e_rend
-import envi.archs.h8 as e_h8
+import envi.archs.arm as arm
 import vivisect
 import platform
-import unittest
+import unittest  
 from envi import IF_RET, IF_NOFALL, IF_BRANCH, IF_CALL, IF_COND
-from envi.archs.h8.regs import *
-from envi.archs.h8.const import *
-from envi.archs.h8.parsers import *
+from envi.archs.arm.regs import *
+from envi.archs.arm.const import *
+from envi.archs.arm.disasm import *
+import binascii   # temporarily included for testing purposes only - allows to print out binary values
 
 
-# OPHEX, VA, repr, flags, emutests
+
+
+''' 
+  This dictionary will contain all commands supported by ARM to test
+  Fields will contain following information:
+  archBitMask, OPHEX, VA, repr, flags, emutests
+'''
+#items commented out are either not yet implimented or raise exceptions due to bugs.
+#all errors found are commented for instruction involved
+#note that cmn, cmp, teq and tst all have s's improperly attached to them for testing.
+#this will be the case until the 's' issue is resolved
+
 instrs = [
-        ('08309fe5', 0xbfb00000, 'ldr r3, [#0xbfb00010]', 0, ()),
-        ('0830bbe5', 0xbfb00000, 'ldr r3, [r11, #0x8]!', 0, ()),
-        ('08309be4', 0xbfb00000, 'ldr r3, [r11], #0x8', 0, ()),
-        ('08301be4', 0xbfb00000, 'ldr r3, [r11], #-0x8', 0, ()),
-        ('02209ae7', 0xbfb00000, 'ldr r2, [r10, r2]', 0, ()),
-        ('02209ae6', 0xbfb00000, 'ldr r2, [r10], r2', 0, ()),
-        ('02203ae7', 0xbfb00000, 'ldr r2, [r10, -r2]!', 0, ()),
-        ('0220bae7', 0xbfb00000, 'ldr r2, [r10, r2]!', 0, ()),
-        ('22209ae7', 0xbfb00000, 'ldr r2, [r10, r2 lsr #32]', 0, ()),
-        ('08309fe5', 0xbfb00000, 'ldr r3, [#0xbfb00010]', 0, ()),
-        ('08309fe5', 0xbfb00000, 'ldr r3, [#0xbfb00010]', 0, ()),
-        ]
-# FIXME: create list of this from IDA flow below - THIS CURRENT DATA IS FOR H8!  NOT ARM/THUMB
+        (0xffff, '674503e0', 0x4560, 'and r4, r3, r7, ror #10', 0, ()),
```

Comment:
Done -

Date: Fri, 3 Jun 2016 11:02:48 -0700
From: notifications@github.com
To: vivisect@noreply.github.com
CC: redheadedrod@hotmail.com; author@noreply.github.com
Subject: Re: [atlas0fd00m/vivisect] Update unit tests for opcodes (#3)

In envi/tests/test_arch_arm.py:

>  instrs = [
> -        ('08309fe5', 0xbfb00000, 'ldr r3, [#0xbfb00010]', 0, ()),
> -        ('0830bbe5', 0xbfb00000, 'ldr r3, [r11, #0x8]!', 0, ()),
> -        ('08309be4', 0xbfb00000, 'ldr r3, [r11], #0x8', 0, ()),
> -        ('08301be4', 0xbfb00000, 'ldr r3, [r11], #-0x8', 0, ()),
> -        ('02209ae7', 0xbfb00000, 'ldr r2, [r10, r2]', 0, ()),
> -        ('02209ae6', 0xbfb00000, 'ldr r2, [r10], r2', 0, ()),
> -        ('02203ae7', 0xbfb00000, 'ldr r2, [r10, -r2]!', 0, ()),
> -        ('0220bae7', 0xbfb00000, 'ldr r2, [r10, r2]!', 0, ()),
> -        ('22209ae7', 0xbfb00000, 'ldr r2, [r10, r2 lsr #32]', 0, ()),
> -        ('08309fe5', 0xbfb00000, 'ldr r3, [#0xbfb00010]', 0, ()),
> -        ('08309fe5', 0xbfb00000, 'ldr r3, [#0xbfb00010]', 0, ()),
> -        ]
> -# FIXME: create list of this from IDA flow below - THIS CURRENT DATA IS FOR H8!  NOT ARM/THUMB
> +        (0xffff, '674503e0', 0x4560, 'and r4, r3, r7, ror #10', 0, ()),

please replace all the 0xffff magic number madness with REV_ALL

and set REV_ALL in const.py


—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub, or mute the thread.